lib/sequent/core/event_store.rb

```diff
@@ -188,14 +190,18 @@ def replay_events_from_cursor(get_events:, block_size: 2000,
         progress = 0
         cursor = get_events.call
         ids_replayed = []
-        cursor.each_row(block_size: block_size).each do |record|
-          event = deserialize_event(record)
-          publish_events([event])
-          progress += 1
-          ids_replayed << record['aggregate_id']
-          if progress % block_size == 0
-            on_progress[progress, false, ids_replayed]
-            ids_replayed.clear
+        cursor.each_row_batch(block_size:).each do |records|
+          events = records.map(&method(:deserialize_event))
+          Sequent.configuration.transaction_provider.transactional do
+            Sequent.configuration.event_publisher.replay_events(events)
+          end
+          records.each do |record|
```

Comment:
This is not really true anymore as this is logging after the fact. We should probably rethink logging